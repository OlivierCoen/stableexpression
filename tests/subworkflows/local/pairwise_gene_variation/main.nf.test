nextflow_workflow {

    name "Test Workflow PAIRWISE_GENE_VARIATION"
    script "subworkflows/local/pairwise_gene_variation/main.nf"
    workflow "PAIRWISE_GENE_VARIATION"
    tag "subworkflow_pairwise_gene_variation"
    tag "workflow"

    test("Should run without failures") {

        when {
            workflow {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/data/pairwise_gene_variation/make_chunks/input/counts.parquet', checkIfExists: true)
                input[0] = ch_counts
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
