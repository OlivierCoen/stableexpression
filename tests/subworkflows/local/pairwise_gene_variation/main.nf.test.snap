{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,40c21df8bc352ec1b76974c35df6be7b"
                ],
                "m_measures": [
                    "m_measures.csv:md5,40c21df8bc352ec1b76974c35df6be7b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-19T14:42:25.315532878"
    }
}